Runtime/Scripts/PersistentConnection.cs

```diff
@@ -19,50 +38,121 @@ public class PersistentConnection : MonoBehaviour
+      {
+        PropagateError(FindCloudletEventTrigger.Error, "performanceSwitchMargin must between (0 to 1.0f)");
+        return false;
       }
+
       if (integration.matchingEngine.sessionCookie == null)
       {
-        FindCloudletEvent findCloudletEvent = new FindCloudletEvent()
-        {
-          trigger = FindCloudletEventTrigger.Error
-        };
-        integration.NewFindCloudletHandler(EdgeEventsStatus.error, findCloudletEvent);
-        Debug.LogError(EdgeEventsError.MissingSessionCookie.ToString());
+        PropagateError(FindCloudletEventTrigger.Error, "Missing SessionCookie");
+        return false;
       }
       if (integration.matchingEngine.EdgeEventsConnection.edgeEventsCookie == null)
       {
-        FindCloudletEvent findCloudletEvent = new FindCloudletEvent()
-        {
-          trigger = FindCloudletEventTrigger.Error
-        };
-        integration.NewFindCloudletHandler(EdgeEventsStatus.error, findCloudletEvent);
-        Debug.LogError(EdgeEventsError.MissingSessionCookie.ToString());
+        PropagateError(FindCloudletEventTrigger.Error, "Missing EdgeEvents Cookie");
+        return false;
       }
-      
-      integration.matchingEngine.EdgeEventsReceiver += HandleServerReceivedEvents;
-      
+      return true;
+    }
+
+
+    void StartEdgeEvents(MobiledgeXIntegration mxi)
+    {
+      integration = mxi;
+      config = MobiledgeXIntegration.settings.edgeEventsConfig;
+      //Log summary of EdgeEvents
+      Logger.Log("EdgeEvents Config: LatencyTestPort: " + config.latencyTestPort);
+      Logger.Log("EdgeEvents Config: NewFindCloudletEventTriggers: ");
+      foreach (FindCloudletEventTrigger trigger in config.newFindCloudletEventTriggers)
+      {
+        Logger.Log("Trigger : " + trigger.ToString());
+      }
+      Logger.Log("EdgeEvents Config: LatencyThresholdTriggerMs: " + config.latencyThresholdTriggerMs);
+      Logger.Log("EdgeEvents Config: LatencyUpdatePattern: " + config.latencyConfig.updatePattern);
+      Logger.Log("EdgeEvents Config: LatencyMaxNoUpdates: " + config.latencyConfig.maxNumberOfUpdates);
+      Logger.Log("EdgeEvents Config: LatencyUpdateIntervalSeconds: " + config.latencyConfig.updateIntervalSeconds);
+      Logger.Log("EdgeEvents Config: LocationUpdatePattern: " + config.locationConfig.updatePattern);
+      Logger.Log("EdgeEvents Config: LocationMaxNoUpdates: " + config.locationConfig.maxNumberOfUpdates);
+      Logger.Log("EdgeEvents Config: LocationUpdateIntervalSeconds: " + config.locationConfig.updateIntervalSeconds);
+
+      integration.matchingEngine.EdgeEventsReceiver += HandleReceivedEvents;
       EdgeEventsConnection connection = integration.matchingEngine.EdgeEventsConnection;
-      
+      bool valid = ValidateConfigs();
+      if (!valid)
+      {
+        return;
+      }
       location = LocationService.RetrieveLocation();
       AppPort appPort = integration.GetAppPort(LProto.Tcp, config.latencyTestPort == 0 ? 0 : config.latencyTestPort);
```

Comment:
the ternary operator is unnecessary. can just be:

`AppPort appPort = integration.GetAppPort(LProto.Tcp, config.latencyTestPort);`